src/libraries/System.Security.Cryptography.Cose/tests/CoseKeyTests.SignVerify.cs

```diff
@@ -0,0 +1,585 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#pragma warning disable SYSLIB5006
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Test.Cryptography;
+using Xunit;
+
+using System.Formats.Cbor;
+using System.Linq;
+
+namespace System.Security.Cryptography.Cose.Tests
+{
+    public partial class CoseKeyTests : IClassFixture<CoseTestKeyManager.TestFixture>
+    {
+        private CoseTestKeyManager.TestFixture _keyManagerFixture;
+        private CoseTestKeyManager KeyManager => _keyManagerFixture.KeyManager;
+        private CoseTestKeyManager BadKeyManager => _keyManagerFixture.BadKeyManager;
+
+        public CoseKeyTests(CoseTestKeyManager.TestFixture keyManagerFixture)
+        {
+            _keyManagerFixture = keyManagerFixture;
+        }
+
+        [ConditionalTheory(typeof(MLDsa), nameof(MLDsa.IsSupported))]
+        [MemberData(nameof(AllMLDsaCoseDraftExamples))]
+        public static void DecodeSign1MLDsaCoseDraftExamples(MLDsaAlgorithm algorithm, byte[] mldsaPkHex, byte[] sign1Hex)
+        {
+            using MLDsa pubKey = MLDsa.ImportMLDsaPublicKey(algorithm, mldsaPkHex);
+
+            CoseSign1Message msg = CoseMessage.DecodeSign1(sign1Hex);
+            Assert.NotNull(msg.Content);
+            Assert.True(msg.VerifyEmbedded(CoseKey.FromKey(pubKey)));
+
+            Assert.Equal(2, msg.ProtectedHeaders.Count);
+            Assert.Equal(0, msg.UnprotectedHeaders.Count);
+
+            if (msg.ProtectedHeaders.TryGetValue(CoseHeaderLabel.Algorithm, out var alg))
+            {
+                Assert.Equal(GetExpectedMLDsaAlgorithm(algorithm), alg.GetValueAsInt32());
+            }
+
+            if (msg.ProtectedHeaders.TryGetValue(CoseHeaderLabel.KeyIdentifier, out var keyIdentifier))
+            {
+                // this doesn't seem to correlate with anything provided in the test vectors therefore we just check the length
+                Assert.Equal(32, keyIdentifier.GetValueAsBytes().Length);
+            }
+        }
+
+        private static int GetExpectedMLDsaAlgorithm(MLDsaAlgorithm algorithm)
+        {
+            if (algorithm == MLDsaAlgorithm.MLDsa44)
+            {
+                return (int)CoseAlgorithm.MLDsa44;
+            }
+            else if (algorithm == MLDsaAlgorithm.MLDsa65)
+            {
+                return (int)CoseAlgorithm.MLDsa65;
+            }
+            else if (algorithm == MLDsaAlgorithm.MLDsa87)
+            {
+                return (int)CoseAlgorithm.MLDsa87;
+            }
+            else
+            {
+                Assert.Fail($"Unrecognized MLDsa algorithm: {algorithm.Name}");
+                return 0;
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(AllKeysAndSign1Implementations))]
+        public void TestSignVerifySingleSignerAllAlgorithms(string keyId, CoseTestSign1 signerImplementation)
```

Comment:
Similarly, this feels like it should be with the existing signing tests, not tests regarding CoseKey.

Reply:
I'm a bit torn here - they're all based on AsymmetricAlgorithm... I'll see if I can fairly quickly do this change or not but will send it separately